lib/core-classes/DeviceService.ts

```diff
@@ -310,6 +318,16 @@ export class DeviceService {
     });
     device._source.assetId = null;
 
+    if(Array.isArray(asset._source.deviceIds)) {
+      const filteredDeviceList = [];
+      for(const linkedDevice of asset._source.deviceIds) {
```

Comment:
Nitpicking: You can use [Array.splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) instead

Reply:
I'm not sure that can help me in this situation.

Reply:
If I understand well, you want to remove the `deviceId` from the asset's device IDs
```
> a = ['a', 'b', 'c']
[ 'a', 'b', 'c' ]
> a.splice(a.indexOf('b'), 1)
[ 'b' ]
> a
[ 'a', 'c' ]
```

Reply:
Ok, I have read function definition to quickly.
I will not do the change because it will cause merge problem with the next PR that have other objects in the list and can't use splice.